src/modules/app.js

```diff
@@ -1,15 +1,66 @@
-const scores = [
-  { id: 1, name: 'Name', score: 100 },
-  { id: 2, name: 'Name', score: 20 },
-  { id: 3, name: 'Name', score: 50 },
-  { id: 4, name: 'Name', score: 78 },
-  { id: 5, name: 'Name', score: 128 },
-
-];
-
-const scoreList = document.getElementById('score-List');
-const saveList = () => {
-  scoreList.innerHTML = scores.map((score) => `<li>${score.name}:${score.score}</li>`).join('\n');
+const refreshButton = document.getElementById('refresh');
+const submitButton = document.getElementById('submit-btn');
+const playerNameInput = document.querySelector('#your-name');
+const scoreInput = document.querySelector('#your-score');
+const baseURL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/NIRId4TfhEE56XcSHFTP/scores/';
+
+const renderScores = (scores) => {
+  const scoreList = document.getElementById('score-List');
+  if (Array.isArray(scores)) {
+    scoreList.innerHTML = scores
+      .map((score) => `<li>${score.user}: ${score.score}</li>`)
+      .join('');
+  } else {
+    scoreList.innerHTML = 'No scores found';
+  }
+};
+
+const refreshScores = async () => {
+  try {
+    const response = await fetch(baseURL);
+    if (response.ok) {
+      const scoresData = await response.json();
+      renderScores(scoresData.result);
+    } else {
+      throw new Error('Error fetching scores');
+    }
+  } catch (error) {
+    /* eslint-disable no-console */
+    console.error('Error fetching scores:', error);
+  }
+};
+
+const submitScore = async () => {
+  const playerName = playerNameInput.value;
+  const score = scoreInput.value;
+  const data = { user: playerName, score };
+
+  try {
+    const response = await fetch(baseURL, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (response.ok) {
+      playerNameInput.value = '';
+      scoreInput.value = '';
+      refreshScores();
+    } else {
+      throw new Error('Error submitting score');
+    }
+  } catch (error) {
+    /* eslint-disable no-console */
+    console.error('Error submitting score:', error);
```

Comment:
- [ ] [OPTIONAL] I understand that most times we just console log the error because we cant leave the block empty, but you can throw new error or return error